### ISSUE TITLE
[Keras import] Fully support Lambda layers.

### ISSUE THREAD

**Issue by maxpumperla** [deeplearning4j/deeplearning4j#4704]
#### Issue Description

We have have an OKish workaround for supporting Lambda layers in individual cases, see here:

https://github.com/deeplearning4j/deeplearning4j/blob/4a25300f1452d656b15686cbb224f28c28aa70c1/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/configurations/Keras2ModelConfigurationTest.java#L44-L46

In general, a Keras Lambda layer accepts any lambda function (any expression over backend operations) as forward pass. To support this, we need to understand these functions upon import. One way to go may be to _dump out these lambda functions as tensorflow graphs_ that we import into a layer/cell/module/whatever.

Still a long way to go, just putting this out here.

**Comment by ItsTehStory** [comment]
@maxpumperla could you give me a link to a doc on how I can write my own layer? I read your comments on your referenced issues and didn't quite understand.
OR, is there a way to import TF computational graphs instead of the keras model (but this isn't for this issue, just wondering)

**Comment by maxpumperla** [comment]
@ItsTehStory sorry, I missed your message. Lambda layer support is now inbound and will soon be available in snapshots.

The way to implement simple Lambda layers in DL4J these days is by extending `SameDiffLambdaLayer`, as in the following test:

https://github.com/deeplearning4j/deeplearning4j/blob/dac07c5adc40ed369abe494377afb637d40f980a/deeplearning4j/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/e2e/KerasLambdaTest.java#L51-L57

That test also shows you how to import Keras Lambda layers!

**Comment by ItsTehStory** [comment]
That's really cool! I found a workaround to the problem I had, but that's always good to know for the future!

**Comment by maxpumperla** [comment]
yeah, I think this is a super useful feature for advanced users.Also note that you can import TF models as well, see e.g. here: 

https://github.com/deeplearning4j/deeplearning4j/blob/f86c35095659fc8f66b8e605ccf5739a2b59ac9b/nd4j/nd4j-backends/nd4j-tests/src/test/java/org/nd4j/imports/TensorFlowImportTest.java#L167

**Comment by ItsTehStory** [comment]
I'm no advanced user though :') I was trying to import a python trained model (mask-rcnn) with some extra layers (one lamba layer) into dl4j to speed up the inference. We ended up realising that we would need to rewrite the whole utils/class/everything to feed the model correctly and that would slow the development (ironic right). I did get the .pb imported into dl4j with the help of your gitter, so that's another cool thing!
